snc.sh

```diff
@@ -238,47 +238,43 @@ function create_pvs() {
 # This follows https://blog.openshift.com/enabling-openshift-4-clusters-to-stop-and-resume-cluster-vms/
 # in order to trigger regeneration of the initial 24h certs the installer created on the cluster
 function renew_certificates() {
+    # This random_string is created by installer and added to each resource type,
+    # in installer side also variable name is kept as `random_string`
+    # so to maintain consistancy, we are also using random_string here.
+    random_string=$(sudo virsh list --all | grep -oP "(?<=${CRC_VM_NAME}-).*(?=-master-0)")
+    if [ -z $random_string ]; then
+        echo "Could not find virtual machine created by snc.sh"
+        exit 1;
+    fi
+    VM_PREFIX=${CRC_VM_NAME}-${random_string}
+
+    sudo virsh shutdown ${VM_PREFIX}-master-0
+    # Wait till instance started successfully
+    until sudo virsh domstate ${VM_PREFIX}-master-0 | grep shut; do
+        echo " ${VM_PREFIX}-master-0 still running"
+        sleep 3
+    done
+
     # Enable the network time sync and set the clock back to present on host
     sudo date -s '1 day'
     sudo timedatectl set-ntp on
 
-    # Create the unsupported cert rotation config for a day (validity will increase by 30 times so it will become 30days)
-    ${OC} create -n openshift-config configmap unsupported-cert-rotation-config --from-literal='base=86400s'
-
-    # Delete the pods to have cert rotation
-    ${OC} delete pods --all -A --force --grace-period=0
-
-    # Wait till the api server again started
-     while ! ${OC} get nodes >/dev/null 2>&1; do
-         sleep 3
-     done
-     echo "API server is up"
-
-    # Force the rotation for secrets
-    ${OC} get secret -A -o json | ${JQ} -r '.items[] | select(.metadata.annotations."auth.openshift.io/certificate-not-after" | .!=null and fromdateiso8601<='$( date --date='+1year' +%s )') | "-n \(.metadata.namespace) \(.metadata.name)"' | \
-        xargs -n3 ${OC} patch secret -p='{"metadata": {"annotations": {"auth.openshift.io/certificate-not-after": null}}}'
+    sudo virsh start ${VM_PREFIX}-master-0
+    # Wait till ssh connection available
+    until ${SSH} core@api.${CRC_VM_NAME}.${BASE_DOMAIN} -- "exit 0" >/dev/null 2>&1; do
+        echo " ${VM_PREFIX}-master-0 still booting"
+        sleep 2
+    done
```

Comment:
No timeout?

Reply:
we don't have it in create disk also, will try to add it once extract it as part of common functions.